server/session.py

```diff
@@ -495,22 +496,29 @@ def set_protocol_handlers(self, ptuple):
             'masternode.announce.broadcast':
             self.masternode_announce_broadcast,
             'masternode.subscribe': self.masternode_subscribe,
+            'masternode.list': self.masternode_list,
+            'masternode.info': self.masternode_info,
         })
 
+    async def notify_masternodes_async(self):
+        for masternode in self.mns:
+            status = await self.daemon.masternode_list(['status', masternode])
+            self.send_notification('masternode.subscribe',
+                                    [masternode, status.get(masternode)])
+
     def notify(self, height, touched):
         '''Notify the client about changes in masternode list.'''
         result = super().notify(height, touched)
-
-        for masternode in self.mns:
-            status = self.daemon.masternode_list(['status', masternode])
-            self.send_notification('masternode.subscribe',
-                                   [masternode, status.get(masternode)])
+        self.controller.create_task(self.notify_masternodes_async())
         return result
 
+
     # Masternode command handlers
     async def masternode_announce_broadcast(self, signmnb):
         '''Pass through the masternode announce message to be broadcast
-        by the daemon.'''
+        by the daemon.
+
+        signmnb: masternode announce message.'''
```

Comment:
You are not doing any type checking.  It is being passed through to the daemon, so that may be acceptable if the daemon does the type checking.

Reply:
yes, the daemon does all the validations